### DISCUSSION TITLE
New conductors

### DISCUSSION THREAD

**Discussion by Korenovskiy** [ra3xdh/qucs_s#693] [Q&A]
Hi everyone! I need to add some new conductors for my project, the conductors will vary in color. I would like to add two more buttons next to the ordinary conductor. Could sb help with that? Thanks in advance

**Comment by ra3xdh** [comment]
This is technically possible, but not as simple as it seems. You need to perform the following things.

* Add a properties and methods in `wire.cpp` handling wire color
* Provide a GUI controls to set wire color
* Take care of backward compotibility in `Schematic::load` and `Schematic::save` Otherwise an old schematic will not load.

**Reply by Korenovskiy** [reply]
I got it, thanks, the only thing i can't get is how to make changes in GUI? Where can I add the controls?

**Reply by wawuwo** [reply]
>I would like to add two more buttons next to the ordinary conductor

I guess you talking about adding buttons to the toolbar where buttons for "wire", "port", "ground", etc. tools reside. If so, then you have to find the place where this toolbar is created and populated, it's in `initActions` and `initToolBar` member functions in `qucs_init.cpp`. For example, [create](https://github.com/ra3xdh/qucs_s/blob/00c46eadbcff1b5f9950ebd20c1000cfbde72441/qucs/qucs_init.cpp#L471) "insert wire" action, then [add](https://github.com/ra3xdh/qucs_s/blob/00c46eadbcff1b5f9950ebd20c1000cfbde72441/qucs/qucs_init.cpp#L949) it to the toolbar

**Reply by Korenovskiy** [reply]
Okey, thanks. By the way, where is better to edit the code? Should i open it using qt?

**Reply by wawuwo** [reply]
Use the tool which suits you well. I use VSCodium and Emacs. It's just a matter of taste.

**Reply by Korenovskiy** [reply]
okey, thanks. I have question, why does qucs-s use QT, if i don't need to use it to make edits? Just wondering

**Reply by wawuwo** [reply]
You seem to be confusing the QT Framework and tools (like QT Creator) made by QT team. They're not tied to each other, one don't have to use the tools in order to use the framework.
